scene/gui/text_edit.cpp

```diff
@@ -2595,123 +2595,109 @@ void TextEdit::_do_backspace(bool p_word, bool p_all_to_left) {
-					break;
-				}
-
-				deselect(caret_edit_order[j]);
-				carets_to_remove.push_back(caret_edit_order[j]);
-				set_caret_column(0, caret_idx == 0, caret_idx);
-				i = j;
-			}
-			continue;
-		}
-
-		if (p_word) {
-			// Save here as the caret may change when resolving overlaps.
-			int from_column = get_caret_column(caret_idx);
-			int column = get_caret_column(caret_idx);
-			// Check for the case "<word><space><caret>" and ignore the space.
-			// No need to check for column being 0 since it is checked above.
+		} else if (p_word && get_caret_column(caret_idx) > 0) {
+			// In case of "[word][space][caret]" ignore the space.
 			if (is_whitespace(text[get_caret_line(caret_idx)][get_caret_column(caret_idx) - 1])) {
-				column -= 1;
+				prev_column -= 1;
 			}
-			// Get a list with the indices of the word bounds of the given text line.
-			const PackedInt32Array words = TS->shaped_text_get_word_breaks(text.get_line_data(get_caret_line(caret_idx))->get_rid());
-			if (words.is_empty() || column <= words[0]) {
-				// If "words" is empty, meaning no words are left, we can remove everything until the beginning of the line.
-				column = 0;
+
+			PackedInt32Array words = TS->shaped_text_get_word_breaks(text.get_line_data(get_caret_line(caret_idx))->get_rid());
+			if (words.is_empty() || prev_column <= words[0]) {
+				// No words to the left: We can remove everything until the beginning of the line.
+				prev_column = 0;
 			} else {
-				// Otherwise search for the first word break that is smaller than the index from we're currently deleting.
-				for (int c = words.size() - 2; c >= 0; c = c - 2) {
-					if (words[c] < column) {
-						column = words[c];
+				// Only go back one word if there are multiple whitespaces.
+				int chunk_size = get_caret_column(caret_idx) > 1 && is_whitespace(text[get_caret_line(caret_idx)][get_caret_column(caret_idx) - 2]) ? 1 : 2;
+				for (int j = words.size() - chunk_size; j >= 0; j -= chunk_size) {
+					if (words[j] < prev_column) {
+						prev_column = words[j];
 						break;
 					}
 				}
 			}
+		} else if (get_caret_column(caret_idx) == 0) {
+			prev_line = get_caret_line(caret_idx) - 1;
+			prev_column = text[prev_line].length();
+		}
 
-			// Check for any other carets in this range.
-			int overlapping_caret_index = -1;
-			for (int j = i + 1; j < caret_edit_order.size(); j++) {
-				// Check caret and selection in on the right line.
-				if (get_caret_line(caret_edit_order[j]) != get_caret_line(caret_idx) && (!has_selection(caret_edit_order[j]) || get_selection_to_line(caret_edit_order[j]) != get_caret_line(caret_idx))) {
-					break;
-				}
+		// Remove overlapping carets.
+		for (int j = i + 1; j < caret_edit_order.size(); j++) {
+			int prev_caret = caret_edit_order[j];
 
-				// If it has a selection, check it ends with in the range.
-				if ((has_selection(caret_edit_order[j]) && get_selection_to_column(caret_edit_order[j]) < column)) {
+			if (p_word) {
```

Comment:
I'm not sure why these can't have the same longer check here, top one is `p_word`
```
if (has_selection(prev_caret) && get_selection_to_line(prev_caret) == get_caret_line(caret_idx) && get_selection_to_column(prev_caret) >= prev_column && (get_selection_from_line(prev_caret) < get_caret_line(caret_idx) || get_selection_from_column(prev_caret) <= prev_column))
if (has_selection(prev_caret) && get_selection_to_line(prev_caret) == get_caret_line(caret_idx)														  &&  get_selection_from_line(prev_caret) < get_caret_line(caret_idx)                                                         )
```
`prev_column ` will be `0` on `p_all_to_left`.
So the first missing check, should always be true as it would be with it missing.
Similarly, the second check, should always be false as it's an `or`, it's removed it's self from the evaluation?